### PR TITLE
WIP: [ON HOLD] PAT-1567 Update Notification Status Colors

### PR DESCRIPTION
**Pull request by alaa-alshaikh**
### Objective
Update Status bar color when user is having a call otherwise the Status bar color will show the default, refer to [this](https://jira.devops.medable.com/browse/PAT-1567)

### **How To Test:** 

**Branches:**
- PAT: task/PAT-1567_update_Notification_topbar_colors
- Axon: task/PAT-1567_update_Notification_topbar_colors
- Cortex: development
- RS: task/PAT-1567_update_Notification_topbar_colors

- Get the FCM token and Twilio token and send fcm request using postman, while FLASK is closed in your device, you will be able to see the call
- Accept the call you will be directed to OngoingCallActivity, hit back or home screen, then go back to the app, you will be able to see the green status bar in the app, you can end the call from the notification or from the OngoingCallActivity, the status bar color will change to the default in the app

### CODE REVIEW

**Comment by jaimesDSG** [comment]
It's working perfectly for me

File: backbone/src/main/java/org/researchstack/backbone/ui/BaseActivity.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package org.researchstack.backbone.ui
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import org.researchstack.backbone.utils.LocalBroadcaster
+
+open class BaseActivity : AppCompatActivity() {
```
Comment (Gryzor) [inline backbone/src/main/java/org/researchstack/backbone/ui/BaseActivity.kt]: Sorry to be the Devil's advocate here, but I don't like using a BaseActivity/Inheritance for this.

We have Jetpack and there's Lifecycle, so I'd rather have a lifecycle aware component (that can be injected or constructed by hand and set up) that receives the lifecycle events and does what it needs to do. 

Also does this only apply to PinCodeActivity? How are TaskActivity, ConsentTaskActivity, etc. going to do this? Are they going to extend a baseActivity in backbone?

Comment (alaa-alshaikh) [reply backbone/src/main/java/org/researchstack/backbone/ui/BaseActivity.kt]: ConsentTaskActivity and  TaskActivity  extends PinCodeActivity eventually so It going to apply into it.

Comment (Gryzor) [reply backbone/src/main/java/org/researchstack/backbone/ui/BaseActivity.kt]: More reason not to add **another** base activity.

**Comment by pperotti-medable** [comment]
Closing this MR since the counter part in the rest of the projects are closed already. It look like this was left behind.

**Pull request closed without merging**